src/test/components/MarkerChart.test.js

```diff
@@ -380,21 +389,56 @@ describe('MarkerChart', function() {
         getContextMenu,
         findFillTextPosition,
         clickOnMenuItem,
+        clickOnSubMenuItem,
       };
     }
 
     it('displays when right clicking on a marker', () => {
       const {
         rightClick,
         clickOnMenuItem,
+        clickOnSubMenuItem,
         getContextMenu,
         findFillTextPosition,
+        getState,
       } = setupForContextMenus();
 
       rightClick(findFillTextPosition('UserTiming A'));
 
       expect(getContextMenu()).toHaveClass('react-contextmenu--visible');
 
+      clickOnSubMenuItem('From the start of this marker', 0);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 0,
+      });
+
+      clickOnSubMenuItem('From the end of this marker', 0);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 10,
+      });
+
+      clickOnSubMenuItem('From the start of this marker', 1);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 10,
+      });
+
+      clickOnSubMenuItem('From the end of this marker', 1);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 10.0001,
```

Comment:
Is there a floating point error here or is this expected?

Reply:
This is what I got in expected.

Reply:
Yeah, we add `0.0001` in the code for the `end`, so that the marker is present in the range even if it's a "dot" marker.